File: merge_sorted_arrays.cpp
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int i = m-1;
        int j = n-1;
        int k = m + n -1;
        while(i>=0 && j>=0){
            if(nums1[i] > nums2[j]){
                nums1[k] = nums1[i];
                i--;
                k--;
            }
            else{
                nums1[k] = nums2[j];
                j--;
                k--;
                
            }
        }
        
        if(k>=0){
            for(;j>=0 && k>=0;j--,k--)
            nums1[k] = nums2[j];
        }
    }
};

File: Merge-Sorted-Arrays.java
//time-O(m+n)
//space-O(1)
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        if(nums1==null || nums1.length==0 || nums2==null || nums2.length==0) return;
        int p1=m-1, p2=n-1, p3=m+n-1;

        while(p1>=0 && p2>=0 && p3>=0){
            if(nums1[p1] > nums2[p2]){
                nums1[p3] = nums1[p1];
                p1--;
            }
            else{
                nums1[p3] = nums2[p2];
                p2--;
            }
            p3--;
        }

        while(p2 >=0 ){
            nums1[p3] = nums2[p2];
            p2--;
            p3--;
        }

        while(p1 >= 0){
            nums1[p3] = nums1[p1];
            p1--;
            p3--;
        }
    }
}